docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -205,6 +199,52 @@ def every_fifth_asset_c_sensor(context):
         return SkipReason(f"asset_c only materialized {len(asset_events)} times.")
 ```
 
+#### Triggering a partitioned run after an upstream partition is materialized
+
+The <PyObject object="MultiAssetSensorEvaluationContext" method="latest_materialization_records_by_partition"/> method fetches the latest materialization records by partition for a given asset. Using this method, you can trigger downstream partitioned runs when the corresponding partitions of upstream assets are materialized:
+
+```python file=/concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_daily_asset_to_daily_asset endbefore=end_daily_asset_to_daily_asset
+downstream_daily_job = define_asset_job(
+    "downstream_daily_job",
+    AssetSelection.keys("downstream_daily_asset"),
+    partitions_def=daily_partitions_def,
+)
+
+
+@multi_asset_sensor(
+    asset_keys=[AssetKey("upstream_daily_asset")], job=downstream_daily_job
+)
+def trigger_daily_asset_from_daily_asset(context):
+    # Get all partitioned materializations after the cursor
+    materializations_by_partition = context.latest_materialization_records_by_partition(
+        AssetKey("upstream_daily_asset")
+    )
+
+    for partition, materialization in materializations_by_partition.items():
+        downstream_partitions = context.get_downstream_partition_keys(
+            partition,
+            from_asset_key=AssetKey("upstream_daily_asset"),
+            to_asset_key=AssetKey("downstream_daily_asset"),
+        )
+        if downstream_partitions:  # Check that a downstream daily partition exists
+            # Upstream daily partition can only map to at most one downstream daily partition
+            yield downstream_daily_job.run_request_for_partition(
+                downstream_partitions[0], run_key=None
```

Comment:
We can omit `run_key` now

Reply:
I think we now prefer returning a list of RunRequests rather than yielding them one by one, because it makes it clear that runs aren't requested at yield time but rather once the function has returned.